File: golang/1768.go
func mergeAlternately(word1 string, word2 string) string {
    len1 := len(word1)
    len2 := len(word2)
    b := make([]byte,0)

    for i := 0; i < len1; i++ {
        if i >= len2 {
            break
        }
        b = append(b, word1[i], word2[i])
    }
    if len1 > len2 {
        b = append(b, word1[len(word2):]...)
    } 
    if len1 < len2 {
        b = append(b, word2[len(word1):]...)
    }
    return string(b)
}

File: rust/1768.rs
impl Solution {
    pub fn merge_alternately(word1: String, word2: String) -> String {
        let mut iter1 = word1.chars();
        let mut iter2 = word2.chars();
        let mut result = String::new();

        loop {
            match (iter1.next(), iter2.next()) {
                (Some(c1), Some(c2)) => {
                    result.push(c1);
                    result.push(c2);
                }
                (Some(c1), None) => result.push(c1),

                (None, Some(c2)) => result.push(c2),
                (None, None) => break,
            }
        }

        result
    }
}
